sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/ParseDriver.scala

```diff
@@ -203,8 +203,17 @@ case object ParseErrorListener extends BaseErrorListener {
       charPositionInLine: Int,
       msg: String,
       e: RecognitionException): Unit = {
-    val position = Origin(Some(line), Some(charPositionInLine))
-    throw new ParseException(None, msg, position, position)
+    val (start, stop) = offendingSymbol match {
+      case token: CommonToken =>
+        val start = Origin(Some(line), Some(token.getCharPositionInLine))
```

Comment:
seems like computation of `start` can be moved outside ? Only the computation of stop is different between commonToken and non common tokens ?

Also, just for my understanding, can you please briefly explain the difference between the common token and other ones ?

Reply:
It's not exactly the same code, but does it have the same result? Looking OK to me but @rubenfiszel could you comment?

Reply:
From a pure code point of view, it's not equivalent since it is using the token.getCharPositionInline instead of the method arg.

It might be equivalent but that would require an invariant to hold (method getCharPositionInline == token.getCharPositionInLine)  that seems unnecessary since the intent of this specific case is to leverage the informations from the CommonToken directly.

The difference between CommonToken and other types of offending symbols is that it is clear for CommonToken where is the stop.

We use this internally on our fork of spark to get nice language-server-protocol errors that are correctly delimited.

Reply:
@rberenguel thanks for your explanation.